apps/src/applab/appStorage.js

```diff
@@ -233,15 +232,14 @@ var handleDeleteRecord = function(tableName, record, onSuccess, onError) {
     return;
   }
   if (this.status === 404) {
-    onError('error deleting record: could not find record id ' + record.id +
-        ' in table ' + tableName);
+    onSuccess(false);
     return;
   }
   if (this.status < 200 || this.status >= 300) {
     onError('error deleting record: unexpected http status ' + this.status);
     return;
   }
-  onSuccess();
+  onSuccess(true);
```

Comment:
Yeah, I still think having an onComplete and an onError will be less confusing than having an onSuccess (that gets called even if not successful) and an onError.

Is there a 1:1 mapping between calling onComplete with success = false and calling onError?